kornia/feature/keynet.py

```diff
@@ -0,0 +1,315 @@
+import math
+from typing import Dict, List, Optional, Tuple
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from kornia.filters import spatial_gradient
+from kornia.geometry.subpix import NonMaximaSuppression2d
+from kornia.geometry.transform import pyrdown
+
+from .laf import laf_from_center_scale_ori
+from .orientation import PassLAF
+
+keynet_config = {
+
+    'KeyNet_default_config':
+        {
+            # Key.Net Model
+            'num_filters': 8,
+            'num_levels': 3,
+            'kernel_size': 5,
+            # Extraction Parameters
+            'nms_size': 15,
+            'pyramid_levels': 4,
+            'up_levels': 1,
+            'scale_factor_levels': math.sqrt(2),
+            's_mult': 22,
+        },
+}
+
+urls: Dict[str, str] = {}
+urls["keynet"] = "https://github.com/axelBarroso/Key.Net-Pytorch/raw/main/model/weights/keynet_pytorch.pth"
+
+
+class feature_extractor(nn.Module):
```

Comment:
```suggestion
class _FeatureExtractor(nn.Module):
```
I know it's inherited code -- but hurts my eyes :)

Reply:
It would not be compatible with the naming in stored weights...